### PR TITLE
[gitlab] Added print utilities

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

Adds two tasks to inspect / query easily jobs / pipelines.

`-j` or `--jq` adds a `jq` query 

Screenshot 2024-07-03 at 11 07 19

### CODE REVIEW

**Review by amenasria — APPROVED** [review]
LGTM !

File: tasks/gitlab_helpers.py [hunk 1]
```diff
@@ -81,3 +82,45 @@ def create_gitlab_annotations_report(ci_job_id: str, ci_job_name: str):
...
+
+        if jq:
+            jq_flags = "-C" if jq_colors else ""
+            ctx.run(f"echo '{obj.to_json()}' | jq {jq_flags} '{jq}'")
+        else:
+            obj.pprint()
+
+
+@task
+def print_pipeline(ctx, id, repo='DataDog/datadog-agent', jq: str | None = None, jq_colors=True):
```
Comment (FlorentClarret) [inline tasks/gitlab_helpers.py:106]:
```suggestion
def print_pipeline(ctx, ids, repo='DataDog/datadog-agent', jq: str | None = None, jq_colors=True):
```

you can have several `id` IIUC

File: tasks/gitlab_helpers.py [hunk 2]
```diff
@@ -81,3 +82,45 @@ def create_gitlab_annotations_report(ci_job_id: str, ci_job_name: str):
...
+
+
+def print_gitlab_object(
+    get_object: callable, ctx, id, repo='DataDog/datadog-agent', jq: str | None = None, jq_colors=True
+):
+    """
+    Print one or more Gitlab objects in JSON and potentially query them with jq
+    """
+    repo = get_gitlab_repo(repo)
+    ids = [i for i in id.split(",") if i]
```
Comment (FlorentClarret) [inline tasks/gitlab_helpers.py:94]: nit - you could also use [`filter`](https://docs.python.org/3/library/functions.html#filter)

File: tasks/gitlab_helpers.py [hunk 3]
```diff
@@ -81,3 +82,45 @@ def create_gitlab_annotations_report(ci_job_id: str, ci_job_name: str):
...
+    for id in ids:
+        obj = get_object(repo, id)
+
+        if jq:
+            jq_flags = "-C" if jq_colors else ""
+            ctx.run(f"echo '{obj.to_json()}' | jq {jq_flags} '{jq}'")
```
Comment (FlorentClarret) [inline tasks/gitlab_helpers.py:100]: Can we have issues if the json contains `'`s?

File: tasks/gitlab_helpers.py [hunk 4]
```diff
@@ -81,3 +82,45 @@ def create_gitlab_annotations_report(ci_job_id: str, ci_job_name: str):
...
+    """
+
+    def get_pipeline(repo, id):
+        return repo.pipelines.get(id)
+
+    print_gitlab_object(get_pipeline, ctx, id, repo, jq, jq_colors)
+
+
+@task
+def print_job(ctx, id, repo='DataDog/datadog-agent', jq: str | None = None, jq_colors=True):
```
Comment (FlorentClarret) [inline tasks/gitlab_helpers.py:118]: What does `jq` means here? Are these the `jq` parameters?

File: tasks/gitlab_helpers.py (hunk 2 above)
Comment (CelianR) [reply tasks/gitlab_helpers.py:94]: Yep, I think that Ruff replaces filter by this as it is more explicit

File: tasks/gitlab_helpers.py (hunk 4 above)
Comment (CelianR) [reply tasks/gitlab_helpers.py]: `jq` is the JSON query passed to the `jq` executable (example: `.name` to get the name of the resulting object)

File: tasks/gitlab_helpers.py (hunk 3 above)
Comment (CelianR) [reply tasks/gitlab_helpers.py]: Fixed !

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**